### PR TITLE
Favour bool and int over boolean and integer

### LINKED ISSUE #2124: Deprecate integer and boolean types
And suggest to use `int` and `bool` instead. FTR I'm not sure if we should switch those types at runtime as somebody might have overridden built-in type (or is that too far fetched?)

### PR DESCRIPTION
**Pull request by malarzm**
Closes #2124

### CODE REVIEW

File: UPGRADE-2.1.md [hunk 1]
```diff
@@ -5,3 +5,5 @@ the `Doctrine\ODM\MongoDB\Id\IdGenerator` interface.
 
 The `Doctrine\ODM\MongoDB\Mapping\ClassMetadata` class has been marked final. The class will no longer be extendable 
 in ODM 3.0.
+
+`boolean` and `integer` types have been deprecated. Use their shorthand counterparts: `bool` and `int` respectively.
```
Comment (alcaeus) [inline UPGRADE-2.1.md:9]:
```suggestion
The `boolean` and `integer` mapping types have been deprecated. Use their shorthand counterparts: `bool` and `int` respectively.
```

**Comment by malarzm** [comment]
@alcaeus also while I'm here there's also oddly looking `\Doctrine\ODM\MongoDB\Types\IntIdType` type registered under `int_id`. Do you recall anything special about it that couldn't be achieved with `int`? Shall I also deprecate it?

File: lib/Doctrine/ODM/MongoDB/Mapping/ClassMetadata.php [hunk 2]
```diff
@@ -1983,6 +1984,18 @@ public function mapField(array $mapping) : array
         $this->checkDuplicateMapping($mapping);
         $this->typeRequirementsAreMet($mapping);
 
+        $deprecatedTypes = [
+            Type::BOOLEAN => Type::BOOL,
+            Type::INTEGER => Type::INT,
+        ];
+        if (isset($deprecatedTypes[$mapping['type']])) {
+            @trigger_error(sprintf(
+                '"%s" type was deprecated in 2.1 and will be removed in 3.0. Use "%s" instead.',
```
Comment (alcaeus) [inline lib/Doctrine/ODM/MongoDB/Mapping/ClassMetadata.php:1993]: Missed this the first time around: please provide the package name in the message:
```suggestion
                '"%s" type was deprecated in doctrine/mongodb-odm 2.1 and will be removed in 3.0. Use "%s" instead.',
```

**Comment by alcaeus** [comment]
I think deprecating it makes sense. The class itself provides no value, so deprecating it in favour of `int` (and using `IntType` in the mean time) makes sense. I decided to dig a bit and found that there's only one place where we actually use `int_id`: https://github.com/doctrine/mongodb-odm/blob/efdde4cb35a4b34426e92acc51dbc870e1aa05ed/lib/Doctrine/ODM/MongoDB/Mapping/ClassMetadata.php#L1898

I'm not sure why this uses `int_id`, but it may as well use `int`. That could easily be tested by changing it and seeing how much breaks. If that works, we can deprecate `int_id` entirely, just like we deprecate `integer` and `boolean`. Feel free to leave that to a separate PR since this is good to go.

File: lib/Doctrine/ODM/MongoDB/Mapping/ClassMetadata.php (hunk 2 above)
Comment (malarzm) [reply lib/Doctrine/ODM/MongoDB/Mapping/ClassMetadata.php]: Added in const as well 👍

**Pull request merged**